guardian/newsletters-nx
# Use email rendering json

## What does this change?


The email rendering previews are loaded from the json routes defined in https://github.com/guardian/email-rendering/pull/366

The Rendering options Edit form page will include the list of warnings include in the response from email rendering.


## How to test

Run email rendering on port 3010, using t[he branch above](https://github.com/guardian/email-rendering/pull/366).
Run this branch locally with the `USE_LOCAL_EMAIL_RENDERING=true` environment variable.

 - Go to the [edit rendering options page for any newsletter ](http://localhost:4200/launched/rendering-options/muddle-teal)
 - add some automated front section from the UI, that use front paths or collections that don't exisit
 - the warnings will show above the preview explaining why your automated sections do not appear

## How can we measure success?

Easier to use the rendering options form.

## Have we considered potential risks?

Need to keep the type for the response in sync with email rendering.

We could use a zod schema to validate the JSON response from email rendering to allow for the types being out-of sync.

## Images

<img width="1224" alt="Screenshot 2023-09-05 at 14 34 32" src="https://github.com/guardian/newsletters-nx/assets/30567854/e6fe0147-9aee-424c-891e-ad3e0f990257">
